src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1673,20 +1673,33 @@ private BaseListSyntax ParseBaseList()
             {
                 // first type
                 TypeSyntax firstType = this.ParseType();
-                list.Add(_syntaxFactory.SimpleBaseType(firstType));
+
+                ArgumentListSyntax argumentList = null;
+                if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+                {
+                    argumentList = this.ParseParenthesizedArgumentList();
+
+                    if (classOrStructOrInterfaceKeyword.Kind != SyntaxKind.ClassKeyword || !haveParameters)
```

Comment:
>ClassKeyword [](start = 75, length = 12)

The specification does not support arguments when the class keyword is used.

Reply:
> The specification does not support arguments when the class keyword is used.

As clearly stated in the PR title, this PR is focused on parsing base clause for records. Given current implementation, class keyword is required to declare a record. Reconciling that aspect of parsing with the updated spec is out of scope for this PR. 

---
In reply to: [433448280](https://github.com/dotnet/roslyn/pull/44733#discussion_r433448280) [](ancestors = 433448280)